clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -540,6 +540,9 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
   if (!FD->hasAttr<OpenCLKernelAttr>())
     return;
 
+  if (getLangOpts().SYCLIsDevice)
+    Fn->addFnAttr("module-id", Fn->getParent()->getModuleIdentifier());
```

Comment:
Is it reliable way to get unique id for the module? Can't we have two different modules with the same module id?

Reply:
This is a valid concern. It seems hard to come up with 100% reliable ID, so at least a TODO should be added to the code.
Other ideas to make it more reliable: add compilation time,  addresses of Function objects corresponding to kernels in this translation unit.

Not 100% reliability should be OK, though, as the worst result is kernels going to the same module rather than different ones.

Reply:
Added a TODO comment to address this concern later.